### PR TITLE
[MIG] dbfilter_from_header: Migration to 12.0

### PR DESCRIPTION
**Pull request by remytms**
This migrates the module `dbfilter_from_header` to 12.0.

There is a refactor that was proposed to 11.0: #1340.

Should we wait before the refactor is merge into 11.0 or can we migrate this module to 12.0 and than migrate the changes from 11.0 to 12.0 ?

### CODE REVIEW

**Comment by houssine78** [comment]
👍 I've tested it. It works fine on v12

File: dbfilter_from_header/README.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (tbaden) [inline dbfilter_from_header/README.rst]: please use the new split readme [maintainer-tools](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0)

File: dbfilter_from_header/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "summary": "Filter databases with HTTP headers",
+    "version": "12.0.1.0.0",
+    "author": "Therp BV,Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "complexity": "normal",
+    "category": "Tools",
+    "depends": [
+        'web',
+    ],
+    "data": [
```
Comment (tbaden) [inline dbfilter_from_header/__manifest__.py]: if you dont use the tag, just delete it.

File: dbfilter_from_header/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    ],
+    "js": [
```
Comment (tbaden) [inline dbfilter_from_header/__manifest__.py]: if you dont use the tag, just delete it.

File: dbfilter_from_header/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    ],
+    "css": [
```
Comment (tbaden) [inline dbfilter_from_header/__manifest__.py]: if you dont use the tag, just delete it.

File: dbfilter_from_header/override.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    db_filter_hdr = httprequest.environ.get('HTTP_X_ODOO_DBFILTER')
+    if db_filter_hdr:
+        dbs = [db for db in dbs if re.match(db_filter_hdr, db)]
+    return dbs
+
+
+if config.get('proxy_mode') and \
+   'dbfilter_from_header' in config.get('server_wide_modules'):
+    _logger = logging.getLogger(__name__)
+    _logger.info('monkey patching http.db_filter')
```
Comment (tbaden) [inline dbfilter_from_header/override.py:26]: is it really necessary to log?

**Review by tbaden — CHANGES_REQUESTED** [review]
thanks for the work 👍 
I did a code review, just some small changes are needed (see below)

File: dbfilter_from_header/override.py (hunk 5 above)
Comment (houssine78) [reply dbfilter_from_header/override.py:26]: it works this way in the previous versions

**Comment by houssine78** [comment]
@remytms could you have a quick look to make the few requested changes ?

**Comment by remytms** [comment]
Yes, I will fix that as soon as possible. :)

**Comment by remytms** [comment]
@tbaden I have split the readme according to maintainer-tools and fixed empty keys in manifest. :)

File: dbfilter_from_header/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+# Copyright 2018 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "dbfilter_from_header",
+    "summary": "Filter databases with HTTP headers",
+    "version": "12.0.1.0.0",
+    "author": "Therp BV, "
+              "Odoo Community Association (OCA)",
+    "license": "AGPL-3",
```
Comment (tbaden) [inline dbfilter_from_header/__manifest__.py:13]: please add the website tag with the value: https://github.com/OCA/server-tools

**Review by tbaden — CHANGES_REQUESTED** [review]
@remytms thanks for the changes, just something minor.

**Comment by remytms** [comment]
@tbaden website tag added. Thanks for your attentive review.

**Review by tbaden — APPROVED** [review]

**Comment by remytms** [comment]
@houssine78 Can you approve this PR, if it seams good to you ? :)

**Review by houssine78 — APPROVED** [review]

**Pull request merged**